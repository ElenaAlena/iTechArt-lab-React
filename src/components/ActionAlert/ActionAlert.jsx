import Alert from "@material-ui/lab/Alert";

const ActionAlert = ({alert}) =>
  alert ? <Alert severity={alert.type || "info"}>{alert.text}</Alert> : null;
export default ActionAlert;
