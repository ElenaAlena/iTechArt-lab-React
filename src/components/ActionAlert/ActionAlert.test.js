import React from 'react';
import {cleanup, render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ActionAlert from './ActionAlert';

afterEach(cleanup);

it('Alert', () => {
    //info
  const info = renderer.create(
    <ActionAlert alert={{text:'Info'}} />,
  );
  const treeInfo = info.toJSON();
  expect(treeInfo).toMatchSnapshot();

    //error    
  const error = renderer.create(
    <ActionAlert alert={{text:'Error',type:'error'}} />,
  );
  const treeError = error.toJSON();
  expect(treeError).toMatchSnapshot();  

});