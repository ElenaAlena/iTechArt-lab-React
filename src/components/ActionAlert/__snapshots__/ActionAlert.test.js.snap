// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert 1`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
  role="alert"
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  >
    Info
  </div>
</div>
`;

exports[`Alert 2`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
  role="alert"
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  >
    Error
  </div>
</div>
`;
